core/src/main/java/com/github/raulccabreu/redukt/Redukt.kt

```diff
@@ -5,24 +5,20 @@ import com.github.raulccabreu.redukt.middlewares.DebugMiddleware
 import com.github.raulccabreu.redukt.middlewares.Middleware
 import com.github.raulccabreu.redukt.reducers.Reducer
 import com.github.raulccabreu.redukt.states.StateListener
+import java.util.concurrent.ConcurrentHashMap
 import java.util.concurrent.ConcurrentLinkedQueue
 import kotlin.system.measureTimeMillis
 
-class Redukt<T>(state: T, debug: Boolean = false) {
+class Redukt<T>(state: T, val debug: Boolean = false) {
     var state = state
         private set
-    val reducers = ConcurrentLinkedQueue<Reducer<T>>()
-    val middlewares = ConcurrentLinkedQueue<Middleware<T>>()
+    val reducers = ConcurrentHashMap<String, Reducer<T>>()
```

Comment:
my concerns regard the hash map is the order. I guess that hash map not ensure that insertion order right?

Reply:
Yes, it is true, put is it necessary execute in order? In my opinion it is not necessary, but if you see the problem I can add to different "list", the list of reducers/middleware and the list of tag of reducers/middlewares.